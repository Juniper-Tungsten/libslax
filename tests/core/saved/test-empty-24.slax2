version 1.1;

match / {
    <out> {
        var $test = "<test><one>1</one></test>";
        var $res = slax:string-to-xml($test);
        var $r2 = slax:string-to-xml("<a>", "<b>", "<c>", $test, "</c>", "</b>", "</a>");
        
        <one> {
            <res> {
                copy-of $res;
            }
            <r2> {
                copy-of $r2;
            }
        }
        var $ores = slax:xml-to-string($res);
        var $or2 = slax:xml-to-string($r2);
        <two> {
            <ores> $ores;
            <or2> $or2;
        }
        var $xres = slax:string-to-xml($ores);
        var $xr2 = slax:string-to-xml($or2);
        <three> {
            <xres> {
                copy-of $xres;
            }
            <xr2> {
                copy-of $xr2;
            }
        }
    }
}
