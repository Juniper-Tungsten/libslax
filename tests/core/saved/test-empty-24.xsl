<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:slax="http://code.google.com/p/libslax/slax" version="1.0" extension-element-prefixes="slax">
  <xsl:template match="/">
    <out>
      <xsl:variable name="test" select="&quot;&lt;test&gt;&lt;one&gt;1&lt;/one&gt;&lt;/test&gt;&quot;"/>
      <xsl:variable xmlns:slax="http://code.google.com/p/libslax/slax" name="res" select="slax:string-to-xml($test)"/>
      <xsl:variable xmlns:slax="http://code.google.com/p/libslax/slax" name="r2" select="slax:string-to-xml(&quot;&lt;a&gt;&quot;, &quot;&lt;b&gt;&quot;, &quot;&lt;c&gt;&quot;, $test, &quot;&lt;/c&gt;&quot;, &quot;&lt;/b&gt;&quot;, &quot;&lt;/a&gt;&quot;)"/>
      <one>
        <res>
          <xsl:copy-of select="$res"/>
        </res>
        <r2>
          <xsl:copy-of select="$r2"/>
        </r2>
      </one>
      <xsl:variable xmlns:slax="http://code.google.com/p/libslax/slax" name="ores" select="slax:xml-to-string($res)"/>
      <xsl:variable xmlns:slax="http://code.google.com/p/libslax/slax" name="or2" select="slax:xml-to-string($r2)"/>
      <two>
        <ores>
          <xsl:value-of select="$ores"/>
        </ores>
        <or2>
          <xsl:value-of select="$or2"/>
        </or2>
      </two>
      <xsl:variable xmlns:slax="http://code.google.com/p/libslax/slax" name="xres" select="slax:string-to-xml($ores)"/>
      <xsl:variable xmlns:slax="http://code.google.com/p/libslax/slax" name="xr2" select="slax:string-to-xml($or2)"/>
      <three>
        <xres>
          <xsl:copy-of select="$xres"/>
        </xres>
        <xr2>
          <xsl:copy-of select="$xr2"/>
        </xr2>
      </three>
    </out>
  </xsl:template>
</xsl:stylesheet>
