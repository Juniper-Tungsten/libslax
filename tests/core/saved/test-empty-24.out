<?xml version="1.0"?>
<out>
  <one>
    <res>
      <test>
        <one>1</one>
      </test>
    </res>
    <r2>
      <a>
        <b>
          <c>
            <test>
              <one>1</one>
            </test>
          </c>
        </b>
      </a>
    </r2>
  </one>
  <two>
    <ores>&lt;test&gt;
  &lt;one&gt;1&lt;/one&gt;
&lt;/test&gt;</ores>
    <or2>&lt;a&gt;
  &lt;b&gt;
    &lt;c&gt;
      &lt;test&gt;
        &lt;one&gt;1&lt;/one&gt;
      &lt;/test&gt;
    &lt;/c&gt;
  &lt;/b&gt;
&lt;/a&gt;</or2>
  </two>
  <three>
    <xres>
      <test>
  <one>1</one>
</test>
    </xres>
    <xr2>
      <a>
  <b>
    <c>
      <test>
        <one>1</one>
      </test>
    </c>
  </b>
</a>
    </xr2>
  </three>
</out>
